app/services/phone_confirmation/confirmation_session.rb

```diff
@@ -28,7 +28,11 @@ def regenerate_otp
     end
 
     def matches_code?(candidate_code)
-      Devise.secure_compare(candidate_code, code)
+      return Devise.secure_compare(candidate_code, code) if code.nil? || candidate_code.nil?
+
+      crockford_candidate_code = Base32::Crockford.normalize(candidate_code)
```

Comment:
how about just a to_s on these and drop the line above?

Reply:
I wasn't sure on the implications of changing from nil to String for the variables, so I tried to keep the behavior as similar as possible 😬